panic: interface conversion: vector.Any is *vector.NoRip, not *vector.Fusion

```
$ echo -e '{"a":[1,2],"b":{"x":1}}\n{"a":[3],"b":null}' > test.json &&
  super -version &&
  super -dynamic -vam -i fjson -c "UNNEST [...a, b]" test.json

Version: v0.3.0-185-ge8b348f1f

panic: interface conversion: vector.Any is *vector.NoRip, not *vector.Fusion
goroutine 1 [running]:
runtime/debug.Stack()
	/usr/local/opt/go/libexec/src/runtime/debug/stack.go:26 +0x5e
github.com/brimdata/super/runtime/vam/op.(*Catcher).Pull.func1()
	/Users/phil/work/super/runtime/vam/op/mux.go:26 +0x3d
panic({0x10b47240?, 0xd600c954360?})
	/usr/local/opt/go/libexec/src/runtime/panic.go:860 +0x13a
github.com/brimdata/super/vector.Super({0x10e41600, 0xd600c71bee0})
	/Users/phil/work/super/vector/fusion.go:56 +0xbc
github.com/brimdata/super/vector.Apply(0x6, 0xd600c94f800, {0xd600c915b00, 0x2, 0x2})
	/Users/phil/work/super/vector/apply.go:25 +0x4a7
github.com/brimdata/super/runtime/vam/expr.(*ArrayExpr).Eval(0xd600c9158a0, {0x10e41210, 0xd600c915ae0})
	/Users/phil/work/super/runtime/vam/expr/arrayexpr.go:43 +0x31a
github.com/brimdata/super/runtime/vam/op.(*Unnest).Pull(0xd600c91a880, 0x20?)
	/Users/phil/work/super/runtime/vam/op/unnest.go:40 +0xca
github.com/brimdata/super/runtime/vam/op.(*Single).Pull(0xd600c929a40, 0xdd?)
	/Users/phil/work/super/runtime/vam/op/mux.go:173 +0x33
github.com/brimdata/super/runtime/vam/op.(*Catcher).Pull(0xd9d2765?, 0xc0?)
	/Users/phil/work/super/runtime/vam/op/mux.go:29 +0x5c
github.com/brimdata/super/runtime/exec.(*Query).Pull(0x10b49440?, 0x20?)
	/Users/phil/work/super/runtime/exec/query.go:40 +0x3c
github.com/brimdata/super/vector/vio.CopyMux(0xd600c94fd08, {0x5a08f9c0, 0xd600c929a70})
	/Users/phil/work/super/vector/vio/io.go:101 +0x38
github.com/brimdata/super/cmd/super/root.(*Command).Run(0xd600cb30488, {0xd600c6740f0, 0x1, 0x1})
	/Users/phil/work/super/cmd/super/root/command.go:109 +0xb6a
github.com/brimdata/super/pkg/charm.path.run({0xd600c6261a0, 0x1, 0x1}, {0xd600c6740f0, 0x1, 0x0?})
	/Users/phil/work/super/pkg/charm/path.go:11 +0x7b
github.com/brimdata/super/pkg/charm.(*Spec).Exec(0x110dd980, {0xd600c674090, 0x7, 0x7})
	/Users/phil/work/super/pkg/charm/charm.go:74 +0x1fa
main.main()
	/Users/phil/work/super/cmd/super/main.go:40 +0x5b
```

## Details

Repro is with super commit e8b348f.

The repro above is a simplification of the "union" query often performed on GitHub Archive data as part of benchmarks.

```
$ curl -s -O https://data.gharchive.org/2023-02-08-0.json.gz &&
  gunzip 2023-02-08-0.json.gz &&
  super -dynamic -vam -i fjson -f csup -o 2023-02-08-0.csup 2023-02-08-0.json &&
  super -dynamic -vam -i csup -f csv -c "
FROM '2023-02-08-0.csup'
| UNNEST [...payload.pull_request.assignees, payload.pull_request.assignee]
| WHERE this IS NOT NULL
| AGGREGATE count() BY assignee:=login
| ORDER BY count DESC
| LIMIT 5;"

panic: interface conversion: vector.Any is *vector.NoRip, not *vector.Fusion
goroutine 60 [running]:
github.com/brimdata/super/vector.Super({0xcbde600, 0x28fe6279e50})
	/Users/phil/work/super/vector/fusion.go:56 +0xbc
github.com/brimdata/super/vector.Apply(0x6, 0x28fe5e45cb0, {0x28fe5ebb200, 0x2, 0x2})
	/Users/phil/work/super/vector/apply.go:25 +0x4a7
github.com/brimdata/super/vector.Apply-range1(...)
	/Users/phil/work/super/vector/apply.go:41
github.com/brimdata/super/vector.Apply.rip.func1(...)
	/Users/phil/work/super/vector/apply.go:70
github.com/brimdata/super/vector.Apply(0x6, 0x28fe5e45cb0, {0x28fe5ebb1a0, 0x2, 0x2})
	/Users/phil/work/super/vector/apply.go:40 +0x2b1
github.com/brimdata/super/vector.Apply-range1(...)
	/Users/phil/work/super/vector/apply.go:41
github.com/brimdata/super/vector.Apply.rip.func1(...)
	/Users/phil/work/super/vector/apply.go:70
github.com/brimdata/super/vector.Apply(0x6, 0x28fe5e45cb0, {0x28fe5ebb0a0, 0x2, 0x2})
	/Users/phil/work/super/vector/apply.go:40 +0x2b1
github.com/brimdata/super/runtime/vam/expr.(*ArrayExpr).Eval(0x28fe637da80, {0xcbde210, 0x28fe5ebb020})
	/Users/phil/work/super/runtime/vam/expr/arrayexpr.go:43 +0x31a
github.com/brimdata/super/runtime/vam/op.(*Unnest).Pull(0x28fe63aca40, 0x0?)
	/Users/phil/work/super/runtime/vam/op/unnest.go:40 +0xca
github.com/brimdata/super/runtime/vam/op.(*Filter).Pull(0x28fe63b4ff0, 0x0)
	/Users/phil/work/super/runtime/vam/op/filter.go:22 +0x35
github.com/brimdata/super/runtime/vam/op/aggregate.(*Aggregate).Pull(0x28fe63b6160, 0x0?)
	/Users/phil/work/super/runtime/vam/op/aggregate/aggregate.go:60 +0x14c
github.com/brimdata/super/runtime/vam/op.(*combineParent).run(0x28fe63ace40)
	/Users/phil/work/super/runtime/vam/op/combine.go:109 +0x3c
created by github.com/brimdata/super/runtime/vam/op.(*Combine).Pull.func1 in goroutine 1
	/Users/phil/work/super/runtime/vam/op/combine.go:42 +0x35
```

## Claude's take

As I used Claude AI to help create the simplified repro, it gave its assessment of the root cause and fix, should this save anyone time.

># Bug: panic in `vector.Super` with `*NoRip{*Fusion}`
>
>## Root cause
>
>`NoRip` embeds `Any` without overriding `Kind()`, so `(*NoRip{*Fusion}).Kind()` transparently
>returns `KindFusion`. In `Super` (`vector/fusion.go:56`), the guard `vec.Kind() == KindFusion`
>passes for a `*NoRip`-wrapped Fusion, but `vec.(*Fusion)` then panics because the concrete
>type is `*NoRip`.
>
>## Trigger
>
>`ArrayExpr.Eval` calls `AddNoRip(e.Value.Eval(this))` on each non-spread element of an array
>literal. When the field evaluates to a `*Fusion` (which happens for any heterogeneous-typed
>field — e.g. `assignee` is sometimes null, sometimes an object), `AddNoRip` was wrapping it
>directly in `*NoRip`, producing `*NoRip{*Fusion}`. Then `Apply(ApplyRipFusions, ...)` calls
>`Super` on it and panics.
>
>## Fix
>
>Add `vec = Super(vec)` at the top of `AddNoRip` in `vector/apply.go`. This rips any Fusion
>before wrapping, transforming `*Fusion` into its inner `*Dynamic` first. The Dynamic's
>type-specific slots each get their own `*NoRip` wrapper and drive the per-type ripping
>correctly.

--- a/runtime/ztests/expr/array-expr.yaml
+++ b/runtime/ztests/expr/array-expr.yaml
@@ -10,6 +10,7 @@ input: |
   {a:"foo",b:1,c:127.0.0.1}
   {a:"bar",b:2,c:127.0.0.2}
   {a:0,b:"1"::(int64|string),c:null::(int64|string|null)}
+  {a:0,b:1,c:fusion([2]::(null|[int64]),<[int64]>)}
 
 output: |
   [error("missing"),error("quiet"),null]
@@ -20,6 +21,7 @@ output: |
   ["foo",1,127.0.0.1]
   ["bar",2,127.0.0.2]
   [0,"1",null]
+  [0,1,fusion([2]::(null|[int64]),<[int64]>)]
 
 ---
 

--- a/vector/apply.go
+++ b/vector/apply.go
@@ -22,7 +22,9 @@ type NoRip struct {
 func Apply(opt ApplyOpt, eval func(...Any) Any, vecs ...Any) Any {
 	if opt&ApplyRipFusions != 0 {
 		for k, vec := range vecs {
-			vecs[k] = Super(vec)
+			if _, ok := vec.(*NoRip); !ok {
+				vecs[k] = Super(vec)
+			}
 		}
 	}
 	if opt&ApplyRipUnions != 0 {

